src/lib/Microsoft.Health.Events/Telemetry/Exceptions/EventHubExceptionTelemetryProcessor.cs

```diff
@@ -23,26 +23,69 @@ public static class EventHubExceptionTelemetryProcessor
             EnsureArg.IsNotNull(exception, nameof(exception));
             EnsureArg.IsNotNull(logger, nameof(logger));
 
-            logger.LogError(exception);
+            var ex = CustomizeException(exception);
+
+            logger.LogError(ex);
 
             if (shouldLogMetric)
             {
-                logger.LogMetric(EventMetrics.HandledException(errorMetricName ?? GetErrorMetricName(exception), ConnectorOperation.Setup), 1);
+                if (ex.Equals(exception))
+                {
+                    logger.LogMetric(
+                        EventMetrics.HandledException(
+                            errorMetricName ?? $"{ErrorType.EventHubError}{EventHubErrorCode.GeneralError}",
+                            ConnectorOperation.Setup),
+                        1);
+                }
+                else
+                {
+                    var processor = new EventHubConfigurationExceptionTelemetryProcessor();
```

Comment:
Can consider saving this instead of creating a new instance each time.

Reply:
Good point, thanks.